enterprise/frontend/src/embedding-sdk/store/auth.ts

```diff
@@ -0,0 +1,159 @@
+import type {
+  EmbeddingSessionToken,
+  FetchRequestTokenFn,
+  SDKConfig,
+} from "embedding-sdk";
+import type { SdkStoreState } from "embedding-sdk/store/types";
+import api from "metabase/lib/api";
+import { createAsyncThunk } from "metabase/lib/redux";
+import { refreshSiteSettings } from "metabase/redux/settings";
+import { refreshCurrentUser } from "metabase/redux/user";
+
+import { getOrRefreshSession } from "./reducer";
+import { getFetchRefreshTokenFn } from "./selectors";
+
+export const initAuth = createAsyncThunk(
+  "sdk/token/INIT_AUTH",
+  async (sdkConfig: SDKConfig, { dispatch }) => {
+    // Setup JWT or API key
+    const isValidJwtConfig =
+      sdkConfig.jwtProviderUri && sdkConfig.jwtProviderUri?.length > 0;
+    const isValidApiKeyConfig =
+      sdkConfig.apiKey && window.location.hostname === "localhost";
+
+    if (isValidJwtConfig) {
```

Comment:
I know I'll get feedback on this so let me answer in advance :awe: : I inlined the setupSdkAuth because it was literally less than 20 lines and it was hiding the `.unwrap` of getOrRefreshSession nested behind 2 function calls.
I also believe this is still small enough to be readable (even more) in a single function